core/src/main/java/org/elasticsearch/cluster/metadata/AliasValidator.java

```diff
@@ -141,7 +142,10 @@ public void validateAliasFilter(String alias, byte[] filter, QueryShardContext q
 
     private static void validateAliasFilter(XContentParser parser, QueryShardContext queryShardContext) throws IOException {
         QueryParseContext queryParseContext = queryShardContext.newParseContext(parser);
-        QueryBuilder queryBuilder = QueryBuilder.rewriteQuery(queryParseContext.parseInnerQueryBuilder(), queryShardContext);
-        queryBuilder.toFilter(queryShardContext);
+        Optional<QueryBuilder> parseInnerQueryBuilder = queryParseContext.parseInnerQueryBuilder();
+        if (parseInnerQueryBuilder.isPresent()) {
+            QueryBuilder queryBuilder = QueryBuilder.rewriteQuery(parseInnerQueryBuilder.get(), queryShardContext);
+            queryBuilder.toFilter(queryShardContext);
```

Comment:
does this mean that we allow empty filters for aliases? If so can we change that?